test/protocol/modules/CustomOracleNAVIssuanceModule.spec.ts

```diff
@@ -0,0 +1,3541 @@
+          units, // Set is valued at 2000 USDC
+          [setup.issuanceModule.address, customOracleNavIssuanceModule.address]
+        );
+        const managerRedemptionHook = await getRandomAddress();
+        const reserveAssets = [setup.usdc.address, setup.weth.address];
+        const managerFeeRecipient = feeRecipient.address;
+        // Set max managerFee to 20%
+        const maxManagerFee = ether(0.2);
+        // Set max premium to 10%
+        const maxPremiumPercentage = ether(0.1);
+        // Set min SetToken supply required
+        const minSetTokenSupply = ether(1);
+
+        navIssuanceSettings = {
+          managerIssuanceHook,
+          managerRedemptionHook,
+          setValuer: setValuerAddress,
+          reserveAssets,
+          feeRecipient: managerFeeRecipient,
+          managerFees,
+          maxManagerFee,
+          premiumPercentage,
+          maxPremiumPercentage,
+          minSetTokenSupply,
+        } as CustomOracleNAVIssuanceSettings;
+
+        await customOracleNavIssuanceModule.initialize(setToken.address, navIssuanceSettings);
+        // Approve tokens to the controller
+        await setup.weth.approve(setup.controller.address, ether(100));
+        await setup.usdc.approve(setup.controller.address, usdc(1000000));
+        await setup.wbtc.approve(setup.controller.address, bitcoin(1000000));
+        await setup.dai.approve(setup.controller.address, ether(1000000));
+
+        // Seed with 2 supply
+        await setup.issuanceModule.connect(owner.wallet).initialize(setToken.address, ADDRESS_ZERO);
+        await setup.issuanceModule.connect(owner.wallet).issue(setToken.address, ether(2), owner.address);
+
+        // Issue with 1k USDC
+        issueQuantity = usdc(1000);
+
+        await setup.usdc.approve(customOracleNavIssuanceModule.address, issueQuantity);
+
+        subjectSetToken = setToken.address;
+        subjectReserveAsset = setup.usdc.address;
+        subjectReserveQuantity = issueQuantity;
+        subjectMinSetTokenReceived = ether(0);
+        subjectTo = recipient;
+        subjectCaller = owner;
+      });
+
+      context("when using a custom valuer", () => {
+        before(async () => {
+          managerIssuanceHook = ADDRESS_ZERO;
+          // Set fees to 0
+          managerFees = [ether(0), ether(0)];
+          premiumPercentage = ether(0);
+          const setValuerMock = await deployer.mocks.deployCustomSetValuerMock();
+          // set valued at $500 by the custom set valuer
+          // which values ETH at 200 usd and usdc at 1 usd
+          await setValuerMock.setValuation(setup.usdc.address, ether(370));
+          await setValuerMock.setValuation(setup.weth.address, ether(1.85)); // 370/200
+          setValuerAddress = setValuerMock.address;
+        });
+
+        const subject = () => customOracleNavIssuanceModule.issue(setToken.address, setup.usdc.address, usdc(296), ether("0.8"), owner.address);
```

Comment:
Instead of defining a new `subject` let's just define some new `subjectVariables` in a `beforeEach` block. You could also just use the same `subjectVariable` and use the `getExpectedSetTokenIssueQuantity` function found on line 1309.